limiter.go

```diff
@@ -7,14 +7,16 @@ import (
 
 const minInt = -int(^uint(0)>>1) - 1
 
-// The error returned when the read rate exceeds our specification.
+// ErrRateExceeded is the error returned when the read rate exceeds our specification.
 var ErrRateExceeded = errors.New("Read rate exceeded.")
 
 // Limiter is an interface for a rate limiter.
 // There are a few example limiters included in the package, but feel free to go wild with your own.
 type Limiter interface {
 	// Apply this many bytes to the limiter, return ErrRateExceeded if the defined rate is exceeded.
 	Count(int) error
+	// UntilNext returns a channel that can be used to block until next window
+	UntilNext() <-chan time.Time
```

Comment:
Might be nicer to not add this to the full Limiter interface but just on the concrete type, and change the constructors to return the concrete type reference instead (should have done that originally--accept interfaces, return concrete types)

Reply:
done